src/scripts/main.js

```diff
@@ -1,3 +1,76 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field tbody');
+const container = document.querySelector('.container');
+const [
+  appendRow,
+  removeRow,
+  appendColumn,
+  removeColumn,
+] = document.querySelectorAll('.button');
+
+container.addEventListener('click', {
+  handleEvent(ev) {
+    if (!ev.target.classList.contains('button')) {
+      return;
+    }
+
+    const buttonAction = ev.target.classList[0];
+    const method = capitalized(buttonAction);
+
+    this[method]();
+
+    const rowsCount = table.children.length;
+    const columnsCount = table.lastElementChild.children.length;
+    const maxCells = 10;
+    const minCells = 2;
+
+    appendRow.disabled = (rowsCount >= maxCells);
+    removeRow.disabled = (rowsCount <= minCells);
+    appendColumn.disabled = (columnsCount >= maxCells);
+    removeColumn.disabled = (columnsCount <= minCells);
+  },
+
+  appendRow() {
+    const columnsCount = table.querySelector('tr').children.length;
+    const row = document.createElement('tr');
+
+    for (let i = 1; i <= columnsCount; i++) {
+      const cell = document.createElement('td');
+
+      row.append(cell);
+    }
+
+    table.append(row);
+  },
+
+  appendColumn() {
+    const rows = table.children;
+
+    [...rows].forEach(row => {
+      const cell = document.createElement('td');
+
+      row.append(cell);
+    });
+  },
+
+  removeRow() {
+    table.lastElementChild.remove();
+  },
+
+  removeColumn() {
+    const rows = table.children;
+
+    [...rows].forEach(row => {
+      row.lastElementChild.remove();
+    });
+  },
+});
+
+function capitalized(text) {
```

Comment:
```suggestion
function capitalizeText(text) {
```
Get used to name functions according to these rules:
![image](https://user-images.githubusercontent.com/96409563/211263844-defda47e-ef34-4b93-b8be-a20d1691ac72.png)